daemon/daemon_unix.go

```diff
@@ -290,6 +293,33 @@ func (daemon *Daemon) adaptContainerSettings(hostConfig *containertypes.HostConf
 	return nil
 }
 
+// adaptSharedNamespaceContainer replaces a container name with its container ID
+// when a container shares its namespace by container name.
```

Comment:
Please add a comment so as to clarify that this conversion is needed for supporting rename operation